src/compiler/transformers/declarations.ts

```diff
@@ -725,45 +741,197 @@ export function transformDeclarations(context: TransformationContext) {
             // Literal const declarations will have an initializer ensured rather than a type
             return;
         }
-        const shouldAddImplicitUndefined = node.kind === SyntaxKind.Parameter && resolver.requiresAddingImplicitUndefined(node);
-        if (type && !shouldAddImplicitUndefined) {
-            return visitNode(type, visitDeclarationSubtree, isTypeNode);
+        return typeFromDeclaration(node);
+
+        function inferVariableLikeType<T extends VariableLikeDeclaration>(node: T, addUndefined?: boolean) {

```

Comment:
Do all of these functions need to close over `node` and `ignorePrivate`, or could we refactor them out of `ensureType` to avoid the overhead of creating a closure? `ensureType` is called frequently in declaration emit and closures aren't free, so I'd rather we not pay the cost of we can avoid it.